resources/views/components/forms/posts-form.blade.php

```diff
@@ -0,0 +1,107 @@
+@section('head-imports')
+    <x-head.tinymce-config> </x-head.tinymce-config>
+@endsection
+@props(['categories', 'post'])
+
+<div class="flex flex-col gap-8">
+    <div>
+        <label
+            for="title"
+            class="block input-label"
+        >Titulo do Post</label>
+        <input
+            type="text"
+            name="title"
+            id="title"
+            class="form-input w-96"
+            value="{{ $post->title ?? old('title') }}"
+        />
+        <div>
+            <x-inputs.input-error fieldIdentifier="title" />
+        </div>
+    </div>
+
+    <div>
+        <label
+            for="category"
+            class="block input-label"
+        >Categoria</label>
+        <select
+            name="category_id"
+            id="category"
+            class="form-input w-96"
+        >
+            @if (isset($post))
+                @foreach ($categories as $category)
+                    <option
+                        value="{{ $category->id }}"
+                        {{ $category->id == $post->category_id ? 'selected' : '' }}
+                    >
+                        {{ $category->name }}
+                    </option>
+                @endforeach
+            @else
+                @foreach ($categories as $category)
+                    <option
+                        value="{{ $category->id }}"
+                        {{ old('category_id') == $category->id ? 'selected' : '' }}
+                    >
+                        {{ $category->name }}
+                    </option>
+                @endforeach
+            @endif
```

Comment:
Existe um operador de optional chaining no php `?` pra ti acessar um atributo de um objeto apenas quando existir. Acho que tu pode fazer assim aqui:
```suggestion
                @foreach ($categories as $category)
                    <option
                        value="{{ $category->id }}"
                        {{ (old('category_id') == $category->id || $post?->category_id == $category->id) ? 'selected' : '' }}
                    >
                        {{ $category->name }}
                    </option>
                @endforeach
```

Da pra ti da uma pesquisada sobre [ternay operator](https://stitcher.io/blog/shorthand-comparisons-in-php) e [optional chaining](https://stitcher.io/blog/php-8-nullsafe-operator)